src/lib/holocene/table/paginated-table.svelte

```diff
@@ -0,0 +1,194 @@
+<script lang="ts">
+  import ProgressBar from '../progress-bar.svelte';
+
+  import { page } from '$app/stores';
+
+  import { updateQueryParameters } from '$lib/utilities/update-query-parameters';
+
+  import {
+    MAX_PAGE_SIZE,
+    defaultItemsPerPage,
+    pagination,
+    perPageKey,
+    currentPageKey,
+    options,
+  } from '$lib/stores/pagination';
+
+  import Icon from '$lib/holocene/icon/icon.svelte';
+  import FilterSelect from '$lib/holocene/select/filter-select.svelte';
+
+  type Item = $$Generic;
+
+  export let items: Item[];
+  export let updating: boolean = false;
+  export let perPageLabel: string;
+  export let pageButtonLabel: (page: number) => string;
+  export let nextPageButtonLabel: string;
+  export let previousPageButtonLabel: string;
+
+  $: url = $page.url;
+  $: perPage = url.searchParams.get(perPageKey) ?? String(defaultItemsPerPage);
+  $: currentPage = url.searchParams.get(currentPageKey);
+  $: store = pagination(items, perPage, currentPage);
+
+  // keep the 'page-size' url search param within the supported options
+  $: {
+    if (parseInt(perPage, 10) > parseInt(MAX_PAGE_SIZE, 10)) {
+      updateQueryParameters({
+        parameter: perPageKey,
+        value: MAX_PAGE_SIZE,
+        url,
+      });
+    } else if (!options.includes(perPage)) {
+      updateQueryParameters({
+        parameter: perPageKey,
+        value: defaultItemsPerPage,
+        url,
+      });
+    } else {
+      updateQueryParameters({
+        parameter: perPageKey,
+        value: perPage,
+        url,
+      });
+    }
+  }
+
+  // Keep the 'page' url search param within 1 and the total number of pages
+  $: {
+    if ($store.totalPages && parseInt(currentPage, 10) > $store.totalPages) {
+      updateQueryParameters({
+        parameter: currentPageKey,
+        value: $store.totalPages,
+        url,
+      });
+    } else if (currentPage === null || parseInt(currentPage, 10) < 0) {
+      updateQueryParameters({
+        parameter: currentPageKey,
+        value: '1',
+        url,
+      });
+    } else {
+      updateQueryParameters({
+        parameter: currentPageKey,
+        value: currentPage,
+        url,
+      });
+    }
+  }
+
+  const handlePageChange = (page: number) => {
+    updateQueryParameters({
+      parameter: currentPageKey,
+      value: page,
+      url,
+    });
+  };
+
+  $: {
+    if (perPage) store.jumpToPage(currentPage);
+    if (currentPage) store.adjustPageSize(perPage);
+  }
+</script>
+
+<div class="paginated-table-wrapper">
+  <table class="paginated-table">
+    <thead class="paginated-table-header">
+      <slot name="headers" />
+      {#if updating}
+        <ProgressBar />
+      {/if}
+    </thead>
+    <tbody class="paginated-table-body">
+      <slot visibleItems={$store.items} />
+    </tbody>
+  </table>
+  <div class="paginated-table-controls">
```

Comment:
We might want to set a min width for these so things don't start looking like this on smaller screens

<img width="370" alt="Screenshot 2023-08-02 at 12 22 11 PM" src="https://github.com/temporalio/ui/assets/15069288/86b105e2-9691-4620-9e37-24e57da6ad4e">